### PR TITLE
Changed persistence queries from xml to yaml

### PR DESCRIPTION
**Pull request by Anoukh**

### CODE REVIEW

File: components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            PersistenceStoreConfigs deploymentConfigurations = deploymentConfigProvider
+                    .getConfigurationObject(PersistenceStoreConfigs.class);
+            List<Queries> deploymentQueries = deploymentConfigurations.getQueries();
+            List<Queries> componentQueries;
+            URL url = this.getClass().getClassLoader().getResource("queries.yaml");
+            if (url != null) {
+                PersistenceStoreConfigs componentConfigurations = readYamlContent(url.openStream());
+                componentQueries = componentConfigurations.getQueries();
+            } else {
+                throw new RuntimeException("Unable to load queries.yaml file.");
```
Comment (sajithshn) [inline components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java]: Why are we throwing a runtime exception here? Isn't it a IOException or a ConfigurationException.

Comment (Anoukh) [reply components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java]: Fixed in commit #cda7601

File: components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            throw new QueryMappingNotAvailableException("Unable to load queries.", e);
+        } catch (IOException e) {
+            throw new IOException("Unable to load content from queries.yaml file.", e);
+        }
+        return queries;
+    }
+
+    public String getQuery(String key) {
+        if (!queries.containsKey(key)) {
+            throw new RuntimeException("Unable to find the configuration entry for the key: " + key);
```
[the remark this answers is not in the document]
Comment (Anoukh) [reply components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/query/QueryManager.java]: Fixed in commit cda7601

File: components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/beans/PersistenceConfigurations.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (Anoukh) [reply components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/beans/PersistenceConfigurations.java]: Fixed in commit 3c9c165

File: components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/beans/PersistenceConfigurations.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+package org.wso2.carbon.stream.processor.core.persistence.beans;
+
+import org.wso2.carbon.config.annotation.Configuration;
+
+@Configuration(namespace = "state.persistence", description = "Query configurations for state persistence")
+public class PersistenceConfigurations {
+    private boolean enabled = false;
+    private int intervalInMin = 1;
+    private int revisionsToKeep = 3;
+    private String persistenceStore = "org.wso2.carbon.stream.processor.core.persistence.FileSystemPersistenceStore";
```
Comment (nirmal070125) [inline components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/beans/PersistenceConfigurations.java]: Better to use FileSystemPersistenceStore.class.getName()

Comment (Anoukh) [reply components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/persistence/beans/PersistenceConfigurations.java]: Fixed in commit 8d13bdc

**Pull request merged**